File: all-code/1200-1299/1266minTimeToVisitAllPoints.py

class Solution:
    def minTimeToVisitAllPoints(self, points):
        if len(points) <= 1:
            return 0
        res = 0
        for idx, p in enumerate(points[1:]):
            res += max(abs(p[0]-points[idx][0]), abs(p[1]-points[idx][1]))
        return res





obj = Solution()
print(obj.minTimeToVisitAllPoints( [[1,1],[3,4],[-1,0]]))
print(obj.minTimeToVisitAllPoints( [[3,2],[-2,2]]))

File: c++/src/1000-1999/1266minTimeToVisitAllPoints.cpp
#include "lc_pub.h"

class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int n=points.size();
        int ans=0;
        for (int i=0;i<n-1;i++) {
            int x=points[i][0],y=points[i][1];
            int u=points[i+1][0],v=points[i+1][1];
            int d1=abs(x-u),d2=abs(y-v);
            ans+=min(d1,d2)+abs(d1-d2);
        }
        return ans;
    }
};

int main()
{
    std::cout<<"test let us start! %s" << __cplusplus <<std::endl;
    vector<int> nums{3,6,5,1,8};
    Solution so;
    return 0;
}